mrp_lot_number_propagation/models/mrp_bom_line.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class MrpBomLine(models.Model):
+    _inherit = "mrp.bom.line"
+
+    propagate_lot_number = fields.Boolean(
+        default=False,
+    )
+    display_propagate_lot_number = fields.Boolean(
+        compute="_compute_display_propagate_lot_number"
+    )
+
+    @api.depends(
+        "bom_id.display_lot_number_propagation",
+        "bom_id.lot_number_propagation",
+    )
+    def _compute_display_propagate_lot_number(self):
+        for line in self:
+            line.display_propagate_lot_number = (
+                line.bom_id.display_lot_number_propagation
+                and line.bom_id.lot_number_propagation
+            )
+
+    @api.constrains(
+        "propagate_lot_number",
+        "bom_id.lot_number_propagation",
+        "product_id.tracking",
+        "bom_id.product_tmpl_id.tracking",
+    )
+    def _check_propagate_lot_number(self):
+        """
+        This function should check:
+
+        - if the bom has lot_number_propagation marked, there is one and
+          only one line of this bom with propagate_lot_number marked.
+        - the bom line being marked with lot_number_propagation is of the same
+          tracking type as the finished product
+        """
+        for line in self:
+            lines_to_propagate = line.bom_id.bom_line_ids.filtered(
+                lambda o: o.propagate_lot_number
+            )
```

Comment:
this makes me think that constrain should work mrp.bom level not at line level, the order is responsible for lines, and changes in one line should not trigger changes in another line (they may but that brings complexity)

Reply:
Yes, I should move this to BoM level indeed

Reply:
I couldn't make it working by moving it on BoM, and I prefer to not spend too much time on this ATM so I'll let this part like this.

Reply:
With the last changes, we can't anymore half configure the BoM, constraints have been added also at BoM level (+ products level as well to avoid updates on `tracking` field).